internal/metrics/submitter/main.go

```diff
@@ -0,0 +1,103 @@
+package submitter
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"time"
+
+	"github.com/stackrox/image-prefetcher/internal/metrics/gen"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+type Submitter struct {
+	c      chan *gen.Result
+	done   chan struct{}
+	client gen.MetricsClient
+	logger *slog.Logger
+}
+
+func NewSubmitter(logger *slog.Logger, client gen.MetricsClient) *Submitter {
+	return &Submitter{
+		c:      make(chan *gen.Result, 1),
+		done:   make(chan struct{}),
+		client: client,
+		logger: logger,
+	}
+}
+
+func (s *Submitter) Run(ctx context.Context) {
+	defer func() { s.done <- struct{}{} }()
+	if s.client == nil {
+		for range s.c {
+		}
+		return
+	}
+	hostName, err := os.Hostname()
+	if err != nil {
+		s.logger.WarnContext(ctx, "could not obtain hostname", "error", err)
+		hostName = "unknown"
+	}
+
+	var metrics []*gen.Result
+	for metric := range s.c {
+		metric.Node = hostName
+		s.logger.DebugContext(ctx, "metric received", "metric", metric)
+		metrics = append(metrics, metric)
+	}
+
+	if err = s.submit(ctx, metrics); err == nil {
+		s.logger.InfoContext(ctx, "metrics submitted")
+		return
+	}
```

Comment:
Do we need this block? If we remove initall interval then it could be handled by retry below

Reply:
Looking at the code, I'm not sure this backoff library works well with a zero initial interval...